examples/step-82/step-82.cc

```diff
@@ -0,0 +1,1500 @@
+          (2.0 * p(2) - 6.0 * std::pow(p(2), 2) + 4.0 * std::pow(p(2), 3)) *
+          std::pow(p(0) * (1.0 - p(0)) * p(1) * (1.0 - p(1)), 2);
+      }
+    else
+      Assert(false, ExcNotImplemented());
+
+    return return_gradient;
+  }
+
+
+
+  template <int dim>
+  SymmetricTensor<2, dim>
+  ExactSolution<dim>::hessian(const Point<dim> &p,
+                              const unsigned int /*component*/) const
+  {
+    SymmetricTensor<2, dim> return_hessian;
+
+    if (dim == 2)
+      {
+        return_hessian[0][0] = (2.0 - 12.0 * p(0) + 12.0 * p(0) * p(0)) *
+                               std::pow(p(1) * (1.0 - p(1)), 2);
+        return_hessian[0][1] =
+          (2.0 * p(0) - 6.0 * std::pow(p(0), 2) + 4.0 * std::pow(p(0), 3)) *
+          (2.0 * p(1) - 6.0 * std::pow(p(1), 2) + 4.0 * std::pow(p(1), 3));
+        return_hessian[1][1] = (2.0 - 12.0 * p(1) + 12.0 * p(1) * p(1)) *
+                               std::pow(p(0) * (1.0 - p(0)), 2);
+      }
+    else if (dim == 3)
+      {
+        return_hessian[0][0] =
+          (2.0 - 12.0 * p(0) + 12.0 * p(0) * p(0)) *
+          std::pow(p(1) * (1.0 - p(1)) * p(2) * (1.0 - p(2)), 2);
+        return_hessian[0][1] =
+          (2.0 * p(0) - 6.0 * std::pow(p(0), 2) + 4.0 * std::pow(p(0), 3)) *
+          (2.0 * p(1) - 6.0 * std::pow(p(1), 2) + 4.0 * std::pow(p(1), 3)) *
+          std::pow(p(2) * (1.0 - p(2)), 2);
+        return_hessian[0][2] =
+          (2.0 * p(0) - 6.0 * std::pow(p(0), 2) + 4.0 * std::pow(p(0), 3)) *
+          (2.0 * p(2) - 6.0 * std::pow(p(2), 2) + 4.0 * std::pow(p(2), 3)) *
+          std::pow(p(1) * (1.0 - p(1)), 2);
+        return_hessian[1][1] =
+          (2.0 - 12.0 * p(1) + 12.0 * p(1) * p(1)) *
+          std::pow(p(0) * (1.0 - p(0)) * p(2) * (1.0 - p(2)), 2);
+        return_hessian[1][2] =
+          (2.0 * p(1) - 6.0 * std::pow(p(1), 2) + 4.0 * std::pow(p(1), 3)) *
+          (2.0 * p(2) - 6.0 * std::pow(p(2), 2) + 4.0 * std::pow(p(2), 3)) *
+          std::pow(p(0) * (1.0 - p(0)), 2);
+        return_hessian[2][2] =
+          (2.0 - 12.0 * p(2) + 12.0 * p(2) * p(2)) *
+          std::pow(p(0) * (1.0 - p(0)) * p(1) * (1.0 - p(1)), 2);
+      }
+    else
+      Assert(false, ExcNotImplemented());
+
+    return return_hessian;
+  }
+
+
+
+  // @sect3{Implementation of the <code>BiLaplacianLDGLift</code> class}
+
+  // @sect4{BiLaplacianLDGLift::BiLaplacianLDGLift}
+
+  // In the constructor, we set the polynomial degree of the two finite element
+  // spaces, we associate the corresponding DoF handlers to the triangulation,
+  // and we set the two penalty coefficients.
+  template <int dim>
+  BiLaplacianLDGLift<dim>::BiLaplacianLDGLift(const unsigned int fe_degree,
+                                              const double penalty_jump_grad,
+                                              const double penalty_jump_val)
+    : fe(fe_degree)
+    , dof_handler(triangulation)
+    , fe_lift(FE_DGQ<dim>(fe_degree), dim * dim)
```

Comment:
I think I missed the origin of `dim * dim`. Could you explain the origin. I guess it number of entries in the Hessian?

Reply:
Indeed, it corresponds to the size of the Hessian. The discrete Hessian, denoted `H_h` in the introduction and defined such that it weakly converges to the continuous one `D^2`, consists of three parts: the broken Hessian plus the lifting of the jumps of the function and the jumps of the gradient of the function. All these terms are in `$\mathbb{R}^{d\times d}$`.